Mathlib/Data/List/BigOperators/Basic.lean

```diff
@@ -170,14 +170,12 @@ theorem prod_take_mul_prod_drop : ∀ (L : List M) (i : ℕ), (L.take i).prod *
 
 @[to_additive (attr := simp)]
 theorem prod_take_succ :
-    ∀ (L : List M) (i : ℕ) (p), (L.take (i + 1)).prod = (L.take i).prod * L.nthLe i p
-  | [], i, p => by cases p
-  | h :: t, 0, _ => rfl
-  | h :: t, n + 1, p => by
+    ∀ (L : List M) (i : Fin L.length), (L.take (i + 1)).prod = (L.take i).prod * L.get i
```

Comment:
I think this should be stated with the same LHS as before, but a constructor application after `get`; otherwise the LHS is syntactically less general.

Reply:
done